### PR TITLE
fixed tests and valgrind issues

### PR DESCRIPTION
**Pull request by zalgayya**

### CODE REVIEW

**Review by cc0407 — APPROVED** [review]
Awesome! No warnings or errors, and a clean valgrind.

Good to merge in

File: GlobalScaling/GlobalScaling.c [hunk 1]
```diff
@@ -8,17 +8,27 @@ void globalScaling(){
...
+
+    int negativeScale = 0;
+    if(globalScaleValue < 0){
+        globalScaleValue *= -1;
+        negativeScale = 1;
+    }
  
     float fourthValue;
     while((p = getPoint(i)) != NULL){
+        negativeScale = 0;
```
Comment (andrewlinington) [inline GlobalScaling/GlobalScaling.c:20]: negative scale is always 0 with how this was done

File: GlobalScaling/GlobalScaling.c [hunk 2]
```diff
@@ -8,17 +8,27 @@ void globalScaling(){
...
         multiplyMatrix( p, transformationMatrix );
         
         fourthValue = p->element[3];
-        if( fourthValue  != 1) {
+        if( fourthValue  != 1 || fourthValue != -1) {
             for(int j = 3; j >= 0; j--) {
                 p->element[j] /= fourthValue;
             }
         }
+        if(negativeScale){
```
Comment (andrewlinington) [inline GlobalScaling/GlobalScaling.c:29]: never is run

**Review by andrewlinington — CHANGES_REQUESTED** [review]
i think there may be a small error where i commented could you review it to ensure this code performs what you expect

**Review by aidkvndfj — APPROVED** [review]
Everything looks good and runs well.

**Pull request merged**